lsusb: fix incorrect variable type and unalignments when dumping hid …

naruto-abc opened the pull request:
First issue, I found there is a weird line:
<img width="558" alt="HID Device Descriptor" src="https://user-images.githubusercontent.com/36058930/207830509-36c5e76c-51ba-4b8c-9646-e01db22b4e94.png">
This because an unsigned int type is used to hold a potential negative value. In order to fix it, I change some variable type to int. I think int type is also safe to hold the values without overflow. After fixing it:
<img width="578" alt="bLength" src="https://user-images.githubusercontent.com/36058930/207832427-06ecd419-7243-4c37-b27a-fcc242dd7530.png">


Second issue, an unalignment for string "Report Descriptors":
<img width="710" alt="HID Device Descriptor" src="https://user-images.githubusercontent.com/36058930/207832173-0587cfce-dd13-4473-87ac-b58d6501a9e2.png">
After fixing it:
<img width="400" alt="HID Device Descriptor" src="https://user-images.githubusercontent.com/36058930/207832324-f8956382-eba3-4d54-b757-f645354781b6.png">

gregkh reviewed (changes requested):
Please break up into 2 different commits.

gregkh reviewed (changes requested):
Also please provide more information in the changelog text itself, much like you did in the pull request.

naruto-abc commented:
Hi Greg,

I've split it into two commits and added more explanation for each one.
You can still refer to the screenshots in this pull request, which is more straightforward to show the effects.

gregkh commented:
Thanks for the fixes, looks good.
